simple-schema.js

```diff
@@ -643,6 +673,22 @@ SimpleSchema.prototype.addValidator = SimpleSchema.prototype.validator = functio
   this._validators.push(func);
 };
 
+// Used to match key parts.
+// * 1st group is the root key.
+// * 2nd group is the root key if it is a regular expression.
+// * 3rd group is the last key after a $.
+// * 4rd group is the last regular expression after a $.
+// * 5rd group is the leaf key.
+// * 6rd group is the last regular expression after the root.
+var KEY_REGEX = /^([^.\/]+|(\/[^\/]+\/))(?:\.\$\.([^.\/]+|(\/[^\/]+\/))|\.([^.\/]+|(\/[^\/]+\/)))*$/;
```

Comment:
This seems more complex than necessary. As far as I'm concerned, it's a regex key if it has at least two `/` characters anywhere in it. Am I missing something?